app/services/doc_auth_router.rb

```diff
@@ -198,26 +198,34 @@ def self.client(vendor_discriminator: nil, warn_notifier: nil)
   end
   # rubocop:enable Layout/LineLength
 
-  def self.doc_auth_vendor(discriminator: nil)
+  def self.doc_auth_vendor(
+    discriminator: nil,
+    discriminator_parser: method(:default_discriminator_parser)
```

Comment:
I am struggling a bit today, but this is a bit hard to follow for me. I think I'd rather try to make the specs handle the randomness to keep the implementation simpler if we can.

Reply:
I guess what I've realized by writing this, is that it's not the randomness, it's that we use a SHA to distribute a value in a range, and it's really hard to guess numerically what the SHA of something will be

Reply:
So I feel like the `default_discriminator_parser` method is super clear, super easy to test, but maybe the naming is not clear?

Reply:
(added a few more reviewers so we can get more opinions)

Reply:
Hm, yeah, I don't feel too strongly one way or the other, but it feels like this method and `randomize?` shouldn't need to care too much about the actual distribution of the randomness, and injecting that implementation helps make the tests a bit more predictable.

One thing I still have some concern is confidence in testing the actual randomness of `default_discriminator_parser`, since the current test seems to just cover a single value producing a predetermined output.

@amathews-fs may have thoughts here as well, since this [came up during the initial implementation as well](https://github.com/18F/identity-idp/pull/5296#discussion_r693164038).